news/5248-render-speedups-func-calls

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Speed up rendering by avoiding costly/redundant function calls. (#5248)
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* <news item>
```

Comment:
I'm impartial on whether we should or should not include this news entry (meaning, I don't necessarily consider this a "significant" change).
We can include it if we want to track such changes more obviously or exclude it to reduce noise in the changelog :shrug: .